lib/ansible/modules/cloud/amazon/elb_classic_lb_facts.py

```diff
@@ -231,29 +203,29 @@ def main():
         names={'default': [], 'type': 'list'}
     )
     )
-    module = AnsibleModule(argument_spec=argument_spec,
-                           supports_check_mode=True)
+    module = AnsibleAWSModule(argument_spec=argument_spec,
+                              supports_check_mode=True)
 
-    if not HAS_BOTO:
-        module.fail_json(msg='boto required for this module')
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
+    if not region:
+        module.fail_json(msg="region must be specified")
 
     try:
-        region, ec2_url, aws_connect_params = get_aws_connection_info(module)
-        if not region:
-            module.fail_json(msg="region must be specified")
+        connection = boto3_conn(module,
+                                conn_type='client',
+                                resource='elb',
+                                region=region,
+                                endpoint=ec2_url,
+                                **aws_connect_params)
+    except (botocore.exceptions.PartialCredentialsError, botocore.exceptions.ProfileNotFound) as e:
+        module.fail_json_aws(e, msg="Can't authorize connection. Check your credentials and profile.")
 
-        names = module.params['names']
-        elb_information = ElbInformation(
-            module, names, region, **aws_connect_params)
-
-        ec2_facts_result = dict(changed=False,
-                                elbs=elb_information.list_elbs())
-
-    except BotoServerError as err:
-        module.fail_json(msg="{0}: {1}".format(err.error_code, err.error_message),
-                         exception=traceback.format_exc())
+    try:
+        elbs = list_elbs(connection, module.params.get('names'))
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Failed to get load balancer facts.")
 
-    module.exit_json(**ec2_facts_result)
+    module.exit_json(elb_classic_facts=elbs)
```

Comment:
I assume this changes the results that this module returns? I also note that the module is in preview mode and that might be acceptable.

Reply:
Good catch! No reason for me to be changing this, fixed it.